src/python/pants/rules/core/list_roots.py

```diff
@@ -32,7 +32,7 @@ class Roots(Goal):
     else:
       all_paths.add(f"**/{path}/")
 
-  snapshot = await Get[Snapshot](PathGlobs(include=tuple(all_paths)))
+  snapshot = await Get[Snapshot](PathGlobs(globs=sorted(all_paths)))
```

Comment:
The justification for this change isn't clear -- was this intentional? Is this equivalent to the previous code somehow?

Reply:
Ah yes, I did this because `all_paths` is a set and sets do not have stable ordering between runs. That means we invalidate the cache more than necessary. Generally, sets should be avoided with the engine. `sorted` fixes this.